### DISCUSSION TITLE
Possible to reference an IP list text file to be used instead ip putting them in the telegraf.conf files?

### DISCUSSION THREAD

**Discussion by G0nz0uk** [telegraf/telegraf#1934] [General]
Hello,

Is it possible to point to a IP list file instead of putting all the IP addresses to poll into the various telegraf.conf files?

For example currently it's like this on our Linux server:

`agents = [ "10.116.1.100:161","10.116.1.101:161","10.116.1.102:161" ]`

Can we use something like:

`agents = /etc/telegraf/telegraf.d/ipaddresses.txt`

Thanks

**Comment by MKRhere** [comment]
I think you want https://github.com/influxdata/telegraf, not this repository.

**Comment by luweijun1992** [comment]
Also required, and labels can be set for each device
```
[mappings]
  "udp://192.168.255.254" = { deviceName = "USG6306", deviceRegion = "Xiamen", deviceRole = "Firewall" }
  "udp://192.168.255.2" = { deviceName = "Huawei-ACC_2", deviceRegion = "Xiamen", deviceRole = "ACCESS" }
  "udp://192.168.255.5" = { deviceName = "Huawei-ACC_5", deviceRegion = "Shenzhen", deviceRole = "ServerACCESS" }

[[inputs.snmp]]
  agents = [
    "udp://192.168.255.254",
    "udp://192.168.255.2",
    "udp://192.168.255.5",
    "udp://192.168.255.121",
]
```
